Add before and after transpilation images (#1115)
Qiskit/documentation: docs/transpile/index.mdx

Transpilation is the process of rewriting a given input circuit to match the topology of a specific quantum device, and optimize the circuit instructions for execution on noisy quantum systems.  This documentation covers the tooling and workflows for local transpilation available to all Qiskit users, as well as for the cloud-based [Qiskit transpiler service](qiskit-transpiler-service) available to Premium Plan users.  If you're using primitives and are only interested in the default transpilation options provided by the Qiskit Runtime service, read the [Configure runtime compilation for Qiskit Runtime](../run/configure-runtime-compilation) topic.

> The process of transpilation will take a circuit which contains your instructions:
> 
> <img src="/images/transpile/pre-transpiled.png" alt="A two-qubit quantum circuit containing a Hadamard and two CNOT gates." />
> 
> And transform it such that only instructions available on a chosen backend are used, and optimizes those instructions in order to minimize the effects of noise:

Reviewer: ```suggestion
The process of transpilation takes a circuit that contains your instructions:

<img src="/images/transpile/pre-transpiled.png" alt="A two-qubit quantum circuit containing a Hadamard and two CNOT gates." />

It then transforms the circuit to use only instructions available on a chosen backend, and optimizes those instructions in order to minimize the effects of noise:
```

<img src="/images/transpile/transpiled.png" alt="The same two-qubit quantum circuit after transpilation.  It contains RZ, X, SX, and ECR gates." />

> ----

Reviewer 2: What's this for?

Author: The paragraphs and images together felt a bit cluttered so I added a horizontal line to separate them out.  I can just add in a line break instead.

Reviewer 2: Makes sense. Feel free to keep the line break!
